Implement phone mask

ZachCMP opened the pull request:
https://www.pivotaltracker.com/story/show/185740385

ZachCMP committed:
Implement phone mask

gigxz commented on src/components/elements/input/PhoneInput.tsx:
```diff
@@ -0,0 +1,48 @@
+            mask: '(000) 000-0000',
+          },
+          {
+            mask: '{+}0` (000) 000-0000',
```
Can you update this to allow 2 digits for country code?

gigxz reviewed (changes requested):
Looks like some phone numbers aren't being displayed on the client profile. Can you update that to be more permissive? It can continue to apply formatting, but if the phone string doesn't match the expected legnth/chars, we should still display it as-is. We shouldn't be hiding any data.

<img width="625" alt="Screen Shot 2023-08-28 at 8 19 39 AM" src="https://github.com/greenriver/hmis-frontend/assets/5333982/89ceb67a-1ad6-4085-aceb-2381a7722042">
<img width="925" alt="Screen Shot 2023-08-28 at 8 19 29 AM" src="https://github.com/greenriver/hmis-frontend/assets/5333982/7fb712e9-7b3d-43ca-a200-8de8837f7df9">

gigxz commented:
Oh, I just realized that the numbers are actually being persisted in the db with the special characters, like `value: "+1 (802) 999-1111"`. I think we should only store the numeric values (and `+`), and just use the input mask for display purposes. That gives us more flexibility as to how we display them, and makes the data cleaner for export purposes if needed.

Passing the `unmask` prop to the IMaskInput component seems to achieve the desired behavior

ZachCMP committed:
Merge remote-tracking branch 'origin/release-86' into 185740385-phone-input-mask

ZachCMP committed:
Adjust phone input; use imask to display phone numbers too

ZachCMP commented:
@gigxz Addressed your feedback. I set it to allow for 2 country codes and also switched the logic to display phone numbers to use the same IMask formatter as the inputs rather than the regex mechanism that was being used before. That should make the input/display of phone numbers consistent. I also adjusted it so that it only persists the numbers. I had to add some extra logic to that to actually persist the `+` symbol, as it needs to be stripped if the number is 10 digits or less otherwise the base value will match the wrong mask pattern. Hopefully that works for you, otherwise I can try and see if I can tune it a bit more.

ZachCMP committed:
Fix import

gigxz committed:
Be more forgiving when displaying bad data

gigxz reviewed (approved):
This looks good, thanks! It still could be more forgiving to display malformed phone numbers, so I'm pushing a fix for that.
